spec/controllers/books_controller_spec.rb

```diff
@@ -26,19 +51,242 @@
     before do
       @book = create(:book)
     end
-    context 'find result' do
-      it 'book found' do
+    context 'book is existing' do
+      it "should be received book's info" do
         get :show, params: { id: @book.id }
         expect(assigns(:book).attributes).to eql(@book.attributes)
       end
+    end
 
-      it 'book not found' do
+    context 'book is not existing' do
+      before do
         allow(Book).to receive(:find_by).with(anything).and_return(nil)
         get :show, params: { id: @book.id }
+      end
+
+      it 'should be received a nil object' do
         expect(assigns(:book)).to equal(nil)
+      end
+
+      it 'should be received a danger flash' do
+        expect(flash.count).to equal(1)
+        expect(flash[:danger]).to eql(I18n.t('not_found'))
+      end
+
+      it 'should be return root page' do
+        expect(subject).to redirect_to(root_url)
+      end
+    end
+  end
+
+  describe '#new' do
+    context 'after login' do
+      include_context 'logged in'
+      it 'get a instace of Book' do
+        get :new
+        expect(assigns(:book)).to be_a(Book)
+      end
+    end
+
+    context 'before login' do
+      before do
+        get :new
+      end
+      it 'get a danger flash' do
+        expect(flash.count).to equal(1)
+        expect(flash[:danger]).to eql(I18n.t('not_found'))
+      end
+
+      it 'return root page' do
+        expect(subject).to redirect_to(root_url)
+      end
+    end
+  end
+
+  describe '#create' do
+    include_context 'logged in'
+    before do
+      @book_params = create(:book)
+      @book_count = Book.count
+      @image_files = []
+      image_files
+    end
+    context 'with valid book_params' do
+      before do
+        post :create, params: { book: valid_book_params }
+      end
+
+      it 'the number of book is increment by 1' do
+        expect(Book.count).to equal(@book_count + 1)
```

Comment:
Use expect change, not manual increase.
https://relishapp.com/rspec/rspec-expectations/v/2-0/docs/matchers/expect-change